backend/src/federation/client/1_0/FederationClient.ts

```diff
@@ -0,0 +1,51 @@
+import { GraphQLClient, gql } from 'graphql-request'
+import { backendLogger as logger } from '@/server/logger'
+import { Community as DbCommunity } from '@entity/Community'
+
+// eslint-disable-next-line camelcase
+export async function requestGetPublicKey(dbCom: DbCommunity): Promise<string | undefined> {
+  let endpoint = dbCom.endPoint.endsWith('/') ? dbCom.endPoint : dbCom.endPoint + '/'
+  endpoint = `${endpoint}${dbCom.apiVersion}/getPublicKey`
```

Comment:
Why the `getPublicKey`?

Reply:
yes that's possibly correct and will be changed as soon as with issue #2630 the correct graphql endpoint exists and the handshake can be tested and optimized